neuralprophet/utils_plot.py

```diff
@@ -0,0 +1,532 @@
+    if (
+        "quantiles" in components
+        and validator == "plot_components"
+        and len(m.model.quantiles) > 1
+        and forecast_in_focus is None
+    ):
+        if len(m.config_train.quantiles) > 1 and (
+            m.n_forecasts > 1 or m.config_ar.n_lags > 0
+        ):  # rather query if n_forecasts >1 than n_lags>1
+            raise ValueError(
+                "Please specify step_number using the highlight_nth_step_ahead_of_each_forecast function"
+                " for quantiles plotting when autoregression enabled."
+            )
+        for i in range(1, len(m.model.quantiles)):
+            plot_components.append(
+                {
+                    "plot_name": "Uncertainty",
+                    "comp_name": f"yhat1 {round(m.model.quantiles[i] * 100, 1)}%",
+                    "fill": True,
+                }
+            )
+    elif (
+        "uncertainty" in components
+        and validator == "plot_components"
+        and len(m.model.quantiles) > 1
+        and forecast_in_focus is not None
+    ):
+        for i in range(1, len(m.model.quantiles)):
+            plot_components.append(
+                {
+                    "plot_name": "Uncertainty",
+                    "comp_name": f"yhat{forecast_in_focus} {round(m.model.quantiles[i] * 100, 1)}%",
+                    "fill": True,
+                }
+            )
+    if validator == "plot_parameters":
+        if len(additive_future_regressors) > 0:
+            plot_components.append({"plot_name": "Additive future regressor"})
+        if len(multiplicative_future_regressors) > 0:
+            plot_components.append({"plot_name": "Multiplicative future regressor"})
+        if len(lagged_scalar_regressors) > 0:
+            plot_components.append({"plot_name": "Lagged scalar regressor"})
+        if len(additive_events) > 0:
+            data_params = m.config_normalization.get_data_params(df_name)
+            scale = data_params["y"].scale
+            additive_events = [(key, weight * scale) for (key, weight) in additive_events]
+            plot_components.append({"plot_name": "Additive event"})
+        if len(multiplicative_events) > 0:
+            plot_components.append({"plot_name": "Multiplicative event"})
+
+        valid_configuration = {
+            "components_list": plot_components,
+            "additive_future_regressors": additive_future_regressors,
+            "additive_events": additive_events,
+            "multiplicative_future_regressors": multiplicative_future_regressors,
+            "multiplicative_events": multiplicative_events,
+            "lagged_scalar_regressors": lagged_scalar_regressors,
+            "overwriting_unknown_data_normalization": overwriting_unknown_data_normalization,
+            "df_name": df_name,
+        }
+    elif validator == "plot_components":
+        valid_configuration = {
+            "components_list": plot_components,
+        }
+    return valid_configuration
```

Comment:
At some point we should definitely refactor this method... also in parts related to #1004